test/utils.js

```diff
@@ -100,6 +102,39 @@ async function assertHttp(url, status, spec, replacements = []) {
   });
 }
 
+async function assertHttpDom(url, status, spec) {
+  return new Promise((resolve, reject) => {
+    let data = '';
+    http.get(url, (res) => {
+      try {
+        assert.equal(res.statusCode, status);
+      } catch (e) {
+        res.resume();
+        reject(e);
+      }
+
+      res
+        .on('data', (chunk) => {
+          data += chunk;
+        })
+        .on('end', () => {
+          try {
+            if (spec) {
+              const datadom = new JSDOM(data).window.document;
+              const specdom = new JSDOM(fse.readFileSync(path.resolve(__dirname, 'specs', spec)).toString()).window.document;
```

Comment:
Why not just store the html inline? There is `multiline()` to let that work nicely and if we are using promises we could just promise-load the file…